### PR TITLE
fix(region): Ignore forms without accessible name as landmarks

### LINKED ISSUE #723: Rule: 'region' - Do not treat <form> without accessible labels as form landmarks
Treat only form elements with non-empty ```aria-labelledby```, ```aria-label``` and ```title``` as form landmarks

### PR DESCRIPTION
**Pull request by WilcoFiers**
Closes #723

### CODE REVIEW

File: test/checks/navigation/region.js [hunk 1]
```diff
@@ -102,6 +102,56 @@ describe('region', function () {
...
+	});
+
+  it('treats forms with aria label as landmarks', function () {
+    var checkArgs = checkSetup('<form id="target" aria-label="foo"><p>This is random content.</p></form>');
+    assert.isTrue(checks.region.evaluate.apply(checkContext, checkArgs));
+  });
+  it('treats forms with aria label as landmarks', function () {
+    var checkArgs = checkSetup('<div role="form" id="target" aria-label="foo"><p>This is random content.</p></form>');
+    assert.isTrue(checks.region.evaluate.apply(checkContext, checkArgs));
+  });
```
Comment (isner) [inline test/checks/navigation/region.js]: Maybe these two tests should be combined so the output (same name for two consecutive tests) doesn't smell of accidental copy-pasta.

Comment (WilcoFiers) [reply test/checks/navigation/region.js]: Whoops! Yep, sounds good. Not sure how that snuck in.

Comment (isner) [reply test/checks/navigation/region.js]: Same for "treats forms with an empty aria label as not a landmarks" below.

File: lib/checks/navigation/region.js [hunk 2]
```diff
@@ -17,18 +17,30 @@ const implicitLandmarks = landmarkRoles
...
-	if (node.hasAttribute('role')) {
-		return landmarkRoles.includes(node.getAttribute('role').toLowerCase());
+function isLandmark (virtualNode) {
+	var node = virtualNode.actualNode;
+	var explictRole = (node.getAttribute('role') || '').trim().toLowerCase();
+
+	if (explictRole) {
+		if (explictRole === 'form') {
+			return !!aria.labelVirtual(virtualNode);
+		} else {
```
Comment (isner) [inline lib/checks/navigation/region.js]: Small point, but I'd like to see these "else" blocks removed where they simply return the "happy path":
```js
if (explicitRole === 'form') {
  // edge case
  return !!aria.labelVirtual(virtualNode);
}
// common case
return landmarkRoles.includes(explicitRole);
```

**Comment by WilcoFiers** [comment]
@isner This one needs a thumbs up before I can merge, since it's been updated since your review.

**Pull request merged**